src/renderer/VersionUpdateWidget.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+const ipc = window.electron.ipcRenderer;
+
+export default function VersionUpdateWidget() {
+  const openReleaseDownloadUrl = () => {
+    ipc.sendMessage('openURL', [downloadUrl]);
+  };
+
+  const [widgetClass, setWidgetClass] = React.useState('hidden');
+  const [downloadUrl, setDownloadUrl] = React.useState<string>('');
+
+  React.useEffect(() => {
+    ipc.on('updateAvailable', (downloadUrl) => {
+      setWidgetClass('');
+      setDownloadUrl(downloadUrl as string);
+    });
+  }, []);
+
+  return (
```

Comment:
I'm by no means a react expert (or even good with react) but we do this differently elsewhere.

I would have done

```
const [widgetShown, setWidgetShown] = React.useState<boolean>(false);

(...)

{ widgetShown &&
 ... <the TSX>
}
```

See videoButton around line 200 for examples. 

Happy to be convinced this is better, but interested in your thoughts.

Reply:
Yeah, I'm used to working on Vue rather than React, this is implemented in a similar way as described above.